src/lighteval/tasks/templates/multichoice_config.py

```diff
@@ -0,0 +1,112 @@
+from dataclasses import dataclass
+from typing import Callable
+
+from lighteval.metrics.metrics import Metrics
+from lighteval.metrics.utils import Metric
+from lighteval.tasks.lighteval_task import LightevalTaskConfig
+from lighteval.tasks.requests import Doc
+from lighteval.tasks.templates.formatting_utils import capitalize, fix_ending_punct
+from lighteval.tasks.templates.formulation import Formulation, MCFFormulation, build_answers, build_options
+from lighteval.tasks.templates.translation_literals import TRANSLATION_LITERALS
+from lighteval.utils.language import Language
+
+
+@dataclass
+class MCQInput:
+    question: str
+    choices: list[str]
+    gold_idxs: list[int] | int
+    context: str | None = None
+    instruction: str | None = None
+
+
+@dataclass
+class DatasetConfig:
+    hf_repo: str
+    hf_subset: str
+    trust_dataset: bool = False
+
+
+@dataclass
+class SplitConfig:
+    evaluation_split: tuple[str]
+    few_shots_split: str | None = None
+    few_shots_select: str | None = None
```

Comment:
This redefines the [Doc class](https://github.com/huggingface/lighteval/blob/919be47922867e0c7ea1f820dd23e0ad92331df3/src/lighteval/tasks/requests.py#L163). Why is it necessary? If you need to split the Doc class contents, please do it in the requests file.

Reply:
Removed as disccues before.